src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/ConfigurationSetting.cs

```diff
@@ -23,17 +22,19 @@ public ConfigurationSetting(string key, string value, string label = null)
             Label = label;
         }
 
+        public ConfigurationSetting(string key, string value, string label, string contentType = null, ETag eTag = default, DateTimeOffset? lastModified = null, bool? locked = null) : this(key, value, label)
+        {
+            ContentType = contentType;
+            ETag = eTag;
+            LastModified = lastModified;
+            Locked = locked;
+        }
+
         /// <summary>
         /// The primary identifier of a key-value.
         /// The key is used in unison with the label to uniquely identify a key-value.
         /// </summary>
-        public string Key {
-            get => _key;
-            set {
-                if (value == null) throw new ArgumentNullException(nameof(Key));
-                _key = value;
-            }
-        }
+        public string Key { get; set; }
```

Comment:
Why this change?

Reply:
Having validator in this property setter is pretty random. For example, Value is required too and is not being validated. I think we should validate in service calls instead of model setter because required values might be dependent on the context object is used in.

The other thing to think about here is nullable annotations, it might slightly change how we design model types.